Week-2/1.Flight-booking.js

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+function getFullName(firstname, surname, useFormalName, gender) {
+  firstname = firstname.trim();
+  surname = surname.trim();
+  const fullName = `${firstname} ${surname}`;
+
+  if (!firstname || !surname) {
+    return "Invalid Input";
+  }
+
+  if (!useFormalName) {
+    return fullName;
+  }
+
+  if (gender == "male") {
+    return `Lord ${fullName}`;
+  }
+
+  return `Lady ${fullName}`;
```

Comment:
You usually want to return the more common case the last; if you're returning the exceptions (in this case, the formalName), something can usually be improved

Common case here is `fullName` because everything uses that